### DISCUSSION TITLE
How does this differ from discordgo?

### DISCUSSION THREAD

**Discussion by foodornt** [diamondburned/arikawa#428] [Help]
What purpose does discordgo not serve but this library serves?

**Comment by diamondburned** [comment]
There used to be a list in the README but it's been removed and I don't plan on maintaining one. Here's some items off the top of my head:

- More modular packages (`discord`, `api`, `gateway`, etc. compared to just one big package)
- Type-safe snowflakes (`ChannelID`, `MessageID`, etc.)
- Does not forbid user APIs, which is needed for third-party clients
